libr/core/cmd.c

```diff
@@ -1266,6 +1282,239 @@ static int cmd_stdin(void *data, const char *input) {
 	return r_core_run_script (core, "-");
 }
 
+static void load_table(RCore *core, RTable *t, const char *file) {
+	char *data = NULL;
+
+	if (*file == '$') {
+		const char *cdata = r_cmd_alias_get (core->rcmd, file, 1);
+		if (cdata) {
+			data = strdup (cdata + 1);
+		}
+	} else {
+		data = r_file_slurp (file, NULL);
+	}
+	if (!data) {
+		eprintf ("Error: Cannot slurp '%s'\n", file);
+		return;
+	}
+	RListIter *iter;
+	char *line;
+	RList *lines = r_str_split_list (data, "\n", 0);
+	bool expect_header = false;
+	bool expect_rows = false;
+	const char *separator = "|";
+	int ncols = 0;
+	r_list_foreach (lines, iter, line) {
```

Comment:
since the table has a well defined format:
```
<line to determine the separator>
<headers>
<line to determine start of rows>
<rows>
```
it's much easier to read if those things are done separately instead of using one big loop with multiple booleans that completely change the behaviour of the loop. In particular, because it seems you can't really go ahead if any of those pieces is missing from the table.

Reply:
it skips lines that dont match those rules

Reply:
This function seems still very messy to me.